docs/rfc/rfc-020-onboarding-projects.rst

```diff
@@ -0,0 +1,235 @@
+- 2022-04-25: Imported document to tendermint repository. (@tychoish)
+
+Overview
+--------
+
+This document describes a collection of projects that might be good for new
+engineers joining the Tendermint Core team. These projects mostly describe
+features that we'd be very excited to see land in the code base, but that are
+intentionally outside of the critical path of a release on the roadmap, and
+have the following properties that we think make good on-boarding projects:
+
+- require relatively little context for the project or its history beyond a
+  more isolated area of the code.
+
+- provide exposure to different areas of the codebase, so new team members
+  will have reason to explore the code base and gain expeirence with more than
+  one area of the system.
+
+- be of moderate size, thereby avoiding making small trivial or mechanical changes
+  while also avoiding large intractable changes without natural touchpoints or
+  check-ins.
+  
+Projects
+--------
+
+Before diving into one of these projects, definitely check in with your
+on-boarding buddy to make sure that these issues are still relevant, if
+there's any context or known pitfalls, and to confirm a high level approach or
+design (if relevant.) These descriptions that follow provide some basic
+background and attempt to describe the user stories and the potential impact
+of these project.
+
+E2E Test Systems
+~~~~~~~~~~~~~~~~
+
+Tendermint's E2E framework makes it possible to run small test networks with
+different Tendermint configurations, and make sure that the system works. The
+tests run Tendermint in a separate binary, and the system provides some very
+high level protection against making changes that could break Tendermint in
+otherwise difficult to detect ways. 
+
+Working on the E2E system is a good place to get introduced to the Tendermint
+codebase, particularly for developers who are newer to Golang, as the E2E
+system (generator, runner, etc.) is distinct from the rest of Tendermint and
+comparatively quite small, so it may be easier to begin making changes in this
+area. At the same time, because the E2E system exercises *all* of Tendermint,
+work in this area is a good way to get introduced to various components of the
+system. 
+
+Configurable E2E Workloads
+++++++++++++++++++++++++++
+
+All E2E tests use the same workload (e.g. generated transactions, submitted to
+different nodes in the network,) which is emperically tuned to provide a
+gentle but consistent parallel load that all E2E tests can pass. Ideally, the
+workload generator could be configurable to have different shapes of work
+(bursty, different transaction sizes, weighted to different nodes, etc.) and
+even perhaps further parameterized within a basic shape, which would make it
+possible to use our existing test infrastructure to answer different questions
+about the performance or capability of the system.
+
+The work would involve adding a new paramater to the E2E test manifest, and
```

Comment:
It might make sense to think of this as more of a "file version" than a plain old parameter. A new workload configuration might look a lot different.

Or it might not, but framing it as a "new setting" kind of orients the reader toward "oh hey, this is kind of a quick and easy thing" when in fact it's quite an interesting project with moving parts and decisions to make (though it is, as you say, fairly self-contained).

Reply:
I do agree that this would be pretty interesting but I think it might approach being two projects. The first project being to create a sort of 'puppet' node that's easy to program byzantine behavior into and the second piece of the project being actually defining some workloads for the new node.